test/gtest/uct/test_uct_perf.cc

```diff
@@ -17,6 +17,63 @@ extern "C" {
 #define UCT_ARM_PERF_TEST_MULTIPLIER  15
 
 class test_uct_perf : public uct_test, public test_perf {
+public:
+    void test_execute(unsigned flags = 0) {
+        if (has_transport("ugni_udt")) {
+            UCS_TEST_SKIP;
+        }
+
+        /* For SandyBridge CPUs, don't check performance of far-socket devices */
+        std::vector<int> cpus = get_affinity();
+        bool check_perf       = true;
+        size_t max_iter       = std::numeric_limits<size_t>::max();
+
+        if (ucs_arch_get_cpu_model() == UCS_CPU_MODEL_INTEL_SANDYBRIDGE) {
+            for (std::vector<int>::iterator iter = cpus.begin();
+                 iter != cpus.end(); ++iter) {
+                if (!ucs_cpu_is_set(*iter, &GetParam()->local_cpus)) {
+                    UCS_TEST_MESSAGE << "Not enforcing performance on "
+                                        "SandyBridge far socket";
+                    check_perf = false;
+                    break;
+                }
+            }
+        }
+
+        if (has_transport("tcp")) {
+            check_perf = false; /* TODO calibrate expected performance based on transport */
+            max_iter   = 1000lu;
+        }
+
+        /* Run all tests */
+        for (const test_spec *test_iter = tests; test_iter->title != NULL;
+             ++test_iter) {
+            test_spec test = *test_iter;
+
+            if (has_transport("cuda_copy") || has_transport("gdr_copy")) {
+                test.send_mem_type = UCS_MEMORY_TYPE_HOST;
```

Comment:
`test.send_mem_type = UCS_MEMORY_TYPE_HOST;` is always executed, can put it outside the `if`.

Reply:
Buggy practrize for my opinion... tomorrow we can add more cases here